driver-core/src/main/com/mongodb/connection/ClusterSettings.java

```diff
@@ -254,12 +278,30 @@ public Builder applyConnectionString(final ConnectionString connectionString) {
             maxWaitQueueSize(waitQueueMultiple * maxSize);
 
             if (connectionString.getServerSelectionTimeout() != null) {
-                serverSelectionTimeout(connectionString.getServerSelectionTimeout(), TimeUnit.MILLISECONDS);
+                serverSelectionTimeout(connectionString.getServerSelectionTimeout(), MILLISECONDS);
+            }
+            if (connectionString.getLocalThreshold() != null) {
+                localThreshold(connectionString.getLocalThreshold(), MILLISECONDS);
             }
-
             return this;
         }
 
+        private ServerSelector unpackServerSelector(final ServerSelector serverSelector) {
+            if (serverSelector instanceof CompositeServerSelector) {
+                List<ServerSelector> serverSelectors = ((CompositeServerSelector) serverSelector).getServerSelectors();
+                return new CompositeServerSelector(serverSelectors.subList(1, serverSelectors.size()));
+            }
+            return null;
+        }
+
+        private ServerSelector packServerSelector() {
+            ServerSelector latencyMinimizingServerSelector = new LatencyMinimizingServerSelector(localThresholdMS, MILLISECONDS);
+            if (serverSelector == null) {
+                return latencyMinimizingServerSelector;
+            }
+            return new CompositeServerSelector(asList(latencyMinimizingServerSelector, serverSelector));
```

Comment:
The two server selectors should be in the reverse order.  The latency minimizer has to be last.

Reply:
That was stupid of me. I'll have to update the async settings docs to reflect its no longer the final selector.